src/ModelerNode.js

```diff
@@ -0,0 +1,60 @@
+import { g } from 'jointjs';
+
+export default class ModelerNode {
+  constructor(type, nodeRegistry, moddle, $t) {
+    this.type = type;
+    this.definition = nodeRegistry[type].definition(moddle, $t);
+    this.diagram = nodeRegistry[type].diagram(moddle);
+  }
+
+  isType(bpmnType) {
+    return this.bpmnType === bpmnType;
+  }
+
+  get bpmnType() {
+    return this.definition.$type;
+  }
+
+  getSize() {
+    return {
+      width: this.diagram.bounds.width,
+      height: this.diagram.bounds.height,
+    };
+  }
+
+  setPosition({ x, y }) {
+    this.diagram.bounds.x = x;
+    this.diagram.bounds.y = y;
+  }
+
+  getPosition() {
+    return {
+      x: this.diagram.bounds.x,
+      y: this.diagram.bounds.y,
+    };
+  }
+
+  getCenterPosition() {
+    const { x, y } = this.getPosition();
+    const { width, height } = this.getSize();
+    const rectangle = new g.Rect(x, y, width, height);
+
+    return rectangle.center();
+  }
+
+  setCenterOnPoint({ x, y }) {
```

Comment:
I think this is the only function that I would leave as is, i.e. **not** as a javascript setter. The reason is that this method is carrying out action rather than just being a bare getter and setter.